### PR TITLE
fix(org): explicitly clear org query state on sign-out

### PR DESCRIPTION
**Pull request by yb175**
## Summary

Explicitly clear organization query state on successful sign-out instead of relying on unauthorized refetches for cleanup.

Mounted consumers of `useActiveOrganization`, `useActiveMember`, and `useActiveMemberRole` previously depended on a `/sign-out` invalidation flow that triggered protected org endpoint refetches after the session cookie was already cleared.

This change introduces explicit local cleanup of org query atoms immediately after successful sign-out so mounted hooks rerender synchronously with `null` state.

## Changes

* Add explicit org-state cleanup callback on `/sign-out`
* Reset:

  * `activeOrganization`
  * `activeMember`
  * `activeMemberRole`
* Preserve existing organization invalidation behavior for org mutations
* Add regression coverage validating immediate logout cleanup behavior

## Improvements

* Org state now clears deterministically on logout instead of depending on 401-driven cleanup
* Mounted org hooks update immediately without waiting for network failure responses
* Preserves existing org reactivity flows while improving logout state consistency

## Notes

The existing `$activeOrgSignal` invalidation still triggers redundant org refetches after logout due to current proxy invalidation behavior, so 401 requests may still occur. However, org state is now already cleared before those refetches happen.

Related: #9713

---
## Summary by cubic
Explicitly clear organization query state on sign-out in `better-auth` so mounted org hooks flip to null immediately, without relying on 401-driven cleanup. This stabilizes logout and reduces redundant protected org requests.

- **Bug Fixes**
  - Add a `/sign-out` listener callback that calls `clearOrgState` to reset `activeOrganization`, `activeMember`, `activeMemberRole`, and pending/refetching flags.
  - Keep org invalidation for mutations; make `ClientAtomListener.callback` explicitly optional (`| undefined`) to support the new listener.
  - Add regression test confirming immediate null role after logout and that any post-logout 401s do not affect the cleared local state.

<sup>Written for commit 9096e731c9a008db3fc5126dbd9adeeb93ca721d. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/better-auth/better-auth/pull/9716?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: packages/better-auth/src/plugins/organization/routes/crud-members.test.ts [hunk 1]
```diff
@@ -549,6 +549,73 @@ describe("activeMemberRole", async () => {
...
+			// Verify role loads
+			const activeBefore = await waitForAuthQueryAtom(
+				signOutClient.useActiveMemberRole,
+				(value) => value.data?.role === "owner",
+			);
+			expect(activeBefore.data?.role).toBe("owner");
+
+			networkLog.length = 0; // Clear log before logout
+			await signOutClient.signOut();
+			await nextTick();
```
[the remark this answers is not in the document]
Comment (yb175) [reply packages/better-auth/src/plugins/organization/routes/crud-members.test.ts:761]: Added assertions after signOut().

File: packages/better-auth/src/plugins/organization/client.ts [hunk 2]
```diff
@@ -247,10 +277,22 @@ export const organizationClient = <CO extends OrganizationClientOptions>(
...
 				matcher(path) {
 					return path === "/sign-out" || path.startsWith("/organization");
 				},
 				signal: "$activeOrgSignal",
+				callback(path) {
+					// Explicitly clear org-related state immediately after sign-out
+					if (path === "/sign-out" && clearOrgState) {
+						clearOrgState();
+					}
+				},
```
[the remark this answers is not in the document]
Comment (yb175) [reply packages/better-auth/src/plugins/organization/client.ts:295]: This PR only adds explicit local cleanup so org state clears immediately after logout instead of depending on 401-based cleanup. Splitting the listeners would be a separate follow-up change 😊

**Comment by yb175** [comment]
Hi @bytaesu 
Just following up on this PR whenever you get a chance to review it. Thanks! 😁

**Review by ping-maxwell — DISMISSED** [review]

**Comment by ping-maxwell** [comment]
hey @yb175 please fix CI

**Comment by yb175** [comment]
@ping-maxwell CI issue has been fixed. Could you please re-run the checks for this PR? Thanks! 😊

**Comment by yb175** [comment]
Hi @ping-maxwell,

Just following up on this PR. The failing checks weren’t reproducible locally on my branch. Could you please re-run the CI when you get a chance? Thanks.
